activejob/test/cases/logging_test.rb

```diff
@@ -284,6 +284,16 @@ def test_job_error_logging
     end
   end
 
+  def test_job_error_logging_backtrace_cleaner
+    perform_enqueued_jobs do
+      RescueJob.perform_later "other"
+    rescue RescueJob::OtherError
+      assert_match(/rescue_job\.rb:\d+:in .*perform'/, @logger.messages)
```

Comment:
I see you copied what was done above, but this should be an `assert_raise`, otherwise if `RescueJob.perform_later "other"` stop raising the test will pass.

Reply:
Nice catch! Fixed in c77265c57e